test/main.c

```diff
@@ -444,4 +444,23 @@ UTEST(utf8codepoint, data) {
   }
 }
 
+UTEST(utf8codepointsize, size_1) { ASSERT_EQ(1, utf8codepointsize('A')); }
+
+UTEST(utf8codepointsize, size_4) { ASSERT_EQ(4, utf8codepointsize(0x20C78)); }
+
+UTEST(utf8catcodepoint, data) {
+  char buffer[129];
+  memset(buffer, 0, 129);
```

Comment:
I don't like this line, this is required because after `utf8catcodepoint` returns null, we have no idea where to put the null terminator. Any suggestions welcome.